core/modules/compare/components/Product.ts

```diff
@@ -0,0 +1,22 @@
+import Product from '@vue-storefront/store/types/product/Product'
+
+export const CompareProduct = {
+  name: 'CompareProduct',
+  computed: {
+    isOnCompare () : boolean {
+      return !!this.$store.state.compare.items.find(p => p.sku === this.product.sku)
+    }
+  },
+  methods: {
+    addToCompare (product: Product) {
```

Comment:
same here, what is the purpose of adding product from comapre to comapre? It should be a separate component like `AddToCart` or `AddToWishlist`

Reply:
wishlish and cart are also w good benchmarks for this module since functionality is almost the same

Reply:
Created separate component `addToCompare `